crates/hyperswitch_domain_models/src/payment_methods.rs

```diff
@@ -0,0 +1,420 @@
+// {
+//     fn to_encryptable(self) -> FxHashMap<String, Encryption> {
+//         let mut map = FxHashMap::with_capacity_and_hasher(3, Default::default());
+//         self.payment_method_data
+//             .map(|x| map.insert("payment_method_data".to_string(), x));
+//         self.payment_method_billing_address
+//             .map(|x| map.insert("payment_method_billing_address".to_string(), x));
+//         map
+//     }
+
+//     fn from_encryptable(
+//         mut hashmap: FxHashMap<String, crypto::Encryptable<Secret<String>>>,
+//     ) -> common_utils::errors::CustomResult<
+//         EncryptablePaymentMethodReq,
+//         common_utils::errors::ParsingError,
+//     > {
+//         Ok(EncryptablePaymentMethodReq {
+//             payment_method_data: hashmap.remove("payment_method_data"),
+//             payment_method_billing_address: hashmap.remove("payment_method_billing_address"),
+//         })
+//     }
+// }
+
+#[cfg(all(
+    any(feature = "v1", feature = "v2"),
+    not(feature = "payment_methods_v2")
+))]
+#[derive(Clone, Debug)]
+pub struct PaymentMethod {
+    pub customer_id: common_utils::id_type::CustomerId,
+    pub merchant_id: common_utils::id_type::MerchantId,
+    pub payment_method_id: common_utils::id_type::PaymentMethodId,
+    pub accepted_currency: Option<Vec<storage_enums::Currency>>,
+    pub scheme: Option<String>,
+    pub token: Option<String>,
+    pub cardholder_name: Option<Secret<String>>,
+    pub issuer_name: Option<String>,
+    pub issuer_country: Option<String>,
+    pub payer_country: Option<Vec<String>>,
+    pub is_stored: Option<bool>,
+    pub swift_code: Option<String>,
+    pub direct_debit_token: Option<String>,
+    pub created_at: PrimitiveDateTime,
+    pub last_modified: PrimitiveDateTime,
+    pub payment_method: Option<storage_enums::PaymentMethod>,
+    pub payment_method_type: Option<storage_enums::PaymentMethodType>,
+    pub payment_method_issuer: Option<String>,
+    pub payment_method_issuer_code: Option<storage_enums::PaymentMethodIssuerCode>,
+    pub metadata: Option<pii::SecretSerdeValue>,
+    pub payment_method_data: OptionalEncryptableValue,
+    pub locker_id: Option<String>,
+    pub last_used_at: PrimitiveDateTime,
+    pub connector_mandate_details: Option<serde_json::Value>,
+    pub customer_acceptance: Option<pii::SecretSerdeValue>,
+    pub status: storage_enums::PaymentMethodStatus,
+    pub network_transaction_id: Option<String>,
+    pub client_secret: Option<String>,
+    pub payment_method_billing_address: OptionalEncryptableValue,
+    pub updated_by: Option<String>,
+}
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+#[derive(Clone, Debug)]
+pub struct PaymentMethod {
+    pub customer_id: common_utils::id_type::CustomerId,
+    pub merchant_id: common_utils::id_type::MerchantId,
+    pub created_at: PrimitiveDateTime,
+    pub last_modified: PrimitiveDateTime,
+    pub payment_method: Option<storage_enums::PaymentMethod>,
```

Comment:
We should have a separate type for payments domain where  `payment_method` `payment_method_type` `payment_method_data` all these should be mandatory